### PR TITLE
tiltfile: add config.tiltfile_path, config.tiltfile_dir. Fixes https://github.com/tilt-dev/tilt/issues/3507

### PR DESCRIPTION
**Pull request by nicks**
Hello @landism,

Please review the following commits I made in branch nicks/ch9402:

4f0bef1c6d13effb52db129dc422b92c07220796 (2020-09-17 09:58:21 -0400)
tiltfile: add config.tiltfile_path, config.tiltfile_dir. Fixes https://github.com/tilt-dev/tilt/issues/3507

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/tiltfile/config/config_test.go [hunk 1]
```diff
@@ -459,6 +459,34 @@ print(config.tilt_subcommand)
...
+print(config.tiltfile_dir)
+`)
+
+	_, err := f.ExecFile("Tiltfile")
+	require.NoError(t, err)
+
+	require.Equal(t, f.Path()+"\n", f.PrintOutput())
+}
+
 func NewFixture(tb testing.TB, userConfigState model.UserConfigState, tiltSubcommand model.TiltSubcommand) *starkit.Fixture {
```
Comment (landism) [inline internal/tiltfile/config/config_test.go:490]: it'd be good to have tests that check the value of `config.tiltfile_path`/dir from inside a `load`ed Tiltfile

File: internal/tiltfile/config/config.go [hunk 2]
```diff
@@ -81,6 +81,19 @@ func (e *Extension) OnStart(env *starkit.Environment) error {
 	}
 
 	err := env.AddValue("config.tilt_subcommand", starlark.String(e.TiltSubcommand))
+	if err != nil {
+		return err
+	}
+
+	err = env.AddValue("config.tiltfile_path", starlark.String(env.StartPath()))
```
Comment (landism) [inline internal/tiltfile/config/config.go:88]: I think it'd be good to communicate via the name whether this is the root or current tiltfile path. `config.entry_tiltfile_path`? `config.entrypoint_tiltfile_path`?

**Review by landism — APPROVED** [review]

File: internal/tiltfile/config/config_test.go (hunk 1 above)
Comment (nicks) [reply internal/tiltfile/config/config_test.go:490]: what about entry_path / entry_dir? or maybe main_path / main_dir ?

was trying to find analogues to this in python but was not coming up with good ones.

File: internal/tiltfile/config/config.go (hunk 2 above)
Comment (nicks) [reply internal/tiltfile/config/config.go:88]: what about entry_path / entry_dir? or maybe main_path / main_dir ?

was trying to find analogues to this in python but was not coming up with good ones.

Comment (landism) [reply internal/tiltfile/config/config.go:88]: I think python would be `__main__.__file__`. We've already got `__file__`, but not sure we want to go down the road of exposing `__main__` as a thing.

entry_path or main_path both seem fine to me

Comment (nicks) [reply internal/tiltfile/config/config.go]: changed it to main_path. ya, agree that __main__ might be a road we eventually go down, but not sure this is the right time.

File: internal/tiltfile/config/config_test.go (hunk 1 above)
Comment (nicks) [reply internal/tiltfile/config/config_test.go:500]: done

**Pull request merged**